### PR TITLE
Mutli-arch support 

### PR DESCRIPTION
**Pull request by tejal29**
Add multi-arch support for all kaniko images. 

Prior to this PR,  amazing @mattmoor added multi-arch to the executor image via bazel so that tekton users can use kaniko multi-arch images.
The arch specific images did not contain credential binary for `gcr`, `ecr` and `aws`. After 1:1 conversation with Matt, the reason why they were excluded is, tekton already sets up auth prior to kaniko pod is created.

In this PR, 
-  use `docker buildx` in cloudbuild.yaml to create multi-arch images for `executor:debug` as well as `warmer`.
-  publish new, `executor:slim` & executor:$TAG-slim` images which only contains kaniko binary and config files.
-  remove bazel targets and use `docker buildx` to create image for executor like others

To do
- [x] run the cloud build trigger & verify the images.
-  [x] ping @mattmoor to test/verify `executor:slim` images - optional 
-  [x] ping kaniko slack channel to test debug image - optional

### CODE REVIEW

**Comment by mattmoor** [comment]
cc @ImJasonH 

I hadn't seen buildx on cloud build before.  neat!

File: deploy/cloudbuild-release.yaml [hunk 1]
```diff
@@ -1,110 +1,67 @@
-# This cloudbuild is run on the creation of new tags, which should signify releases.
 timeout: 1800s
 
 steps:
+# Set up builder for muti-arch builds.
+- name: 'gcr.io/cloud-builders/docker'
+  args: ['run', '--privileged', 'linuxkit/binfmt:v0.7']
+  id: 'initialize-qemu'
+- name: 'gcr.io/cloud-builders/docker'
+  args: ['buildx', 'create', '--name', 'mybuilder']
```
Comment (priyawadhwa) [inline deploy/cloudbuild-release.yaml:9]: Is `buildx` meant to be `build`?

**Review by priyawadhwa — APPROVED** [review]
LGTM, just one question

File: deploy/cloudbuild-release.yaml (hunk 1 above)
Comment (mattmoor) [reply deploy/cloudbuild-release.yaml:9]: no buildx is how you invoke the buildkit stuff.

Comment (briandealwis) [inline deploy/cloudbuild-release.yaml:9]: I think you could use `create --use --platform ${_DOCKER_BUILDX_PLATFORMS}` here to (a) limit the builders used and (b) to skip the `select-builder` step immediately below.

**Review by briandealwis — COMMENTED** [review]
LGTM

File: deploy/cloudbuild-release.yaml (hunk 1 above)
Comment (tejal29) [reply deploy/cloudbuild-release.yaml:9]: no its `buildx` to create a cross platform builder and then use it to later.

**Comment by tejal29** [comment]
Testing Notes:  Executor Image
```
➜  kaniko git:(fix_cloudbuild_release) ✗ container-diff diff gcr.io/kaniko-project/executor gcr.io/tejal-test/executor:multiarch-test-1 --type=file

-----File-----

These entries have been added to gcr.io/kaniko-project/executor:
FILE         SIZE
/dev         0
/proc        0
/sys         0

These entries have been deleted from gcr.io/kaniko-project/executor:
FILE                                     SIZE
/kaniko/docker-credential-acr-env        8.7M

These entries have been changed between gcr.io/kaniko-project/executor and gcr.io/tejal-test/executor:multiarch-test-1:
FILE                                       SIZE1        SIZE2
/kaniko/executor                           47.2M        47.6M
/kaniko/docker-credential-acr              8.1M         9.9M
/kaniko/docker-credential-ecr-login        8M           9M
/kaniko/docker-credential-gcr              6.2M         8.9M

➜  kaniko git:(fix_cloudbuild_release) ✗ dive gcr.io/tejal-test/executor:multiarch-test-1
Image Source: docker://gcr.io/tejal-test/executor:multiarch-test-1
Fetching image... (this can take a while for large images)
Analyzing image...
Building cache...
```

The manifest generated is of type `manifest.list.v2+json`
image

**Comment by tejal29** [comment]
Testing Notes: `executor:slim` image

image

**Comment by tejal29** [comment]
Testing Notes: Debug image 
```
➜  kaniko git:(fix_cloudbuild_release) ✗ container-diff diff gcr.io/kaniko-project/executor:debug gcr.io/tejal-test/executor:multiarch-test-1-debug --type=file

-----File-----

These entries have been added to gcr.io/kaniko-project/executor:debug:
FILE         SIZE
/dev         0
/proc        0
/sys         0

These entries have been deleted from gcr.io/kaniko-project/executor:debug: None

These entries have been changed between gcr.io/kaniko-project/executor:debug and gcr.io/tejal-test/executor:multiarch-test-1-debug:
FILE                                       SIZE1        SIZE2
/kaniko/executor                           47.2M        47.6M
/kaniko/warmer                             31.5M        31.7M
/kaniko/docker-credential-acr-env          8.7M         13.2M
/kaniko/docker-credential-acr              8.1M         9.9M
/kaniko/docker-credential-ecr-login        8M           9M
/kaniko/docker-credential-gcr              6.2M         8.9M

➜  kaniko git:(fix_cloudbuild_release) ✗
```

**Comment by tejal29** [comment]
Testing Note: Warmer image
```
container-diff diff gcr.io/kaniko-project/warmer gcr.io/tejal-test/warmer:multiarch-test-1 --type=file 

-----File-----

These entries have been added to gcr.io/kaniko-project/warmer:
FILE                                     SIZE
/dev                                     0
/kaniko/docker-credential-acr-env        13.2M
/proc                                    0
/sys                                     0

These entries have been deleted from gcr.io/kaniko-project/warmer: None

These entries have been changed between gcr.io/kaniko-project/warmer and gcr.io/tejal-test/warmer:multiarch-test-1:
FILE                                       SIZE1        SIZE2
/kaniko/warmer                             31.5M        31.7M
/kaniko/docker-credential-acr              8.1M         9.9M
/kaniko/docker-credential-ecr-login        8M           9M
/kaniko/docker-credential-gcr              6.1M         8.9M


```

**Comment by tejal29** [comment]
Images for test
```
gcr.io/tejal-test/executor:multiarch-test-1
gcr.io/tejal-test/warmer:multiarch-test-1-debug
gcr.io/tejal-test/executor:multiarch-test-1-debug
gcr.io/tejal-test/executor:multiarch-test-1-slim
```

**Comment by tejal29** [comment]
Fixed slim images not containing the SSL certs /cc @mattmoor 
```
gcr.io/tejal-test/executor:multiarch-test-2-slim
```

image

**Comment by darewreck54** [comment]
Posted a general question here.  https://github.com/GoogleContainerTools/kaniko/pull/1474#issuecomment-761030255 If you have docker installed already, why even use kaniko at that point?

**Pull request merged**